src/common.rs

```diff
@@ -38,7 +38,8 @@ macro_rules! pid_decl {
     ($typ:ty) => {
         #[doc = include_str!("../md_doc/pid.md")]
         #[derive(Clone, Copy, Debug, Hash, PartialEq, Eq, PartialOrd, Ord)]
-        pub struct Pid(pub(crate) $typ);
+        #[repr(transparent)]
+        pub struct Pid(pub $typ);
```

Comment:
Since `From` is implemented, you should be able to use `into` and get the underlying type. However, `#[repr(transparent)]` is a good idea.

Reply:
I did try to use into to get a usize, but it didn't work for me

Reply:
This is surprising. Do you have a link to your code?

Reply:
![image](https://user-images.githubusercontent.com/14863743/152704466-145412ee-f6e8-4b90-9721-0d9e9a42c9be.png)

```rust
error[E0277]: the trait bound `usize: From<Pid>` is not satisfied
  --> src\main.rs:25:23
   |
25 |     let pid: usize = pid.into();
   |                          ^^^^ the trait `From<Pid>` is not implemented for `usize`
   |
   = help: the following implementations were found:
             <usize as From<NonZeroUsize>>
             <usize as From<bool>>
             <usize as From<u16>>
             <usize as From<u8>>
   = note: required because of the requirements on the impl of `Into<usize>` for `Pid`

For more information about this error, try `rustc --explain E0277`.
error: could not compile `affinity` due to previous error
```

Reply:
`usize::from(pid)` also does not work.

Reply:
`usize` is for windows/non-unix systems. So unless you're running this code on a windows/non-unix system, it's normal.

Reply:
I missed an important part in the `From` documentation:

> Only implement [Into](https://doc.rust-lang.org/nightly/core/convert/trait.Into.html) when targeting a version prior to Rust 1.41 and converting to a type outside the current crate.

So apparently, when it's a foreign type, it doesn't implement `Into` automatically... By adding this, it works:

```rust
impl Into<usize> for Pid {
    fn into(self) -> usize {
        self.0
    }
}
```

Do you mind updating the `pid_decl` macro to generate the `Into` implementation as well please?